### PR TITLE
[Merged by Bors] - chore: split Geometry/Manifold/ContMDiff

### PR DESCRIPTION
**Pull request by grunweg**
At about 2200 lines, this is currently the longest file in the Geometry/Manifolds.

(It also moves the slowly compiling proof of `ContMDiffWithinAt.cle_arrowCongr` out of a common recompilation path.)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Geometry/Manifold/ContMDiff/Defs.lean [hunk 1]
```diff
@@ -0,0 +1,917 @@
...
+See `Basic.lean` for further basic properties of smooth functions between smooth manifolds,
+`NormedSpace.lean` for the equivalence of manifold-smoothness to usual smoothness,
+`Product.lean` for smoothness results related to the product of manifolds and
+`Atlas.lean` for smoothness of atlas members and local structomorphisms.
+
+## Implementation details
+
+Many properties follow for free from the corresponding properties of functions in vector spaces,
+as being `Cⁿ` is a local property invariant under the smooth groupoid. We take advantage of the
+general machinery developed in `local_invariant_properties.lean` to get these properties
```
Comment (sgouezel) [inline Mathlib/Geometry/Manifold/ContMDiff/Defs.lean:39]:
```suggestion
general machinery developed in `LocalInvariantProperties.lean` to get these properties
```

File: Mathlib/Geometry/Manifold/ContMDiff/Defs.lean [hunk 2]
```diff
@@ -0,0 +1,917 @@
...
+automatically. For instance, the fact that being `Cⁿ` does not depend on the chart one considers
+is given by `lift_prop_within_at_indep_chart`.
```
Comment (sgouezel) [inline Mathlib/Geometry/Manifold/ContMDiff/Defs.lean:41]:
```suggestion
is given by `liftPropWithinAt_indep_chart`.
```

File: Mathlib/Geometry/Manifold/ContMDiff/Defs.lean [hunk 3]
```diff
@@ -0,0 +1,917 @@
...
+/-- A function is `n` times continuously differentiable in a manifold if it is continuous
+and, for any pair of points, it is `n` times continuously differentiable in the charts
+around these points. -/
+def ContMDiff (n : ℕ∞) (f : M → M') :=
+  ∀ x, ContMDiffAt I I' n f x
+#align cont_mdiff ContMDiff
+
+/-- Abbreviation for `ContMDiff I I' ⊤ f`.
+Short note to work with these abbreviations: a lemma of the form `cont_mdiff_foo.bar` will
+apply fine to an assumption `smooth_foo` using dot notation or normal notation.
```
Comment (sgouezel) [inline Mathlib/Geometry/Manifold/ContMDiff/Defs.lean:232]:
```suggestion
Short note to work with these abbreviations: a lemma of the form `ContMdiffFoo.bar` will
apply fine to an assumption `SmoothFoo` using dot notation or normal notation.
```

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks for doing this!

File: Mathlib/Geometry/Manifold/ContMDiff/Defs.lean (hunk 3 above)
Comment (grunweg) [reply Mathlib/Geometry/Manifold/ContMDiff/Defs.lean:232]: I went with `ContMDiffFoo`, matching the lemmas below.

**Comment by grunweg** [comment]
Thank you for the fast review.
bors r+

**Pull request closed without merging**